roles/nova-data/tasks/libvirt.yml

```diff
@@ -102,6 +102,33 @@
   when: "'{{ nova.libvirt_type }}' == 'kvm'"
   changed_when: False
 
+- name: generate really unique uuid
+  command: uuidgen -t
+  args:
+    creates: /etc/machine-id
+  register: machine_id
+
+- name: write machine-id for really uniqe uuid
+  copy:
+    dest: /etc/machine-id
+    content: "{{ machine_id.stdout + '\n'}}"
+    force: no
+  when: machine_id | changed
```

Comment:
I've tested this for a bit. the `creates` worked pre-2.0 as well. It works well.